lib/asset-rev.js

```diff
@@ -24,9 +25,19 @@ function AssetRev(inputTree, options) {
   this.ignore = options.ignore;
   this.description = options.description;
 
+  // first pass - excludes replaceable source code
+  this.exclude = exclude.slice();
+  this.replaceExtensions.forEach(function(ext) { this.exclude.push('**/*.' + ext); }, this);
   var fingerprintTree = Fingerprint(inputTree, this);
+  var assetRewrite = UseRev(fingerprintTree, this);
 
-  return UseRev(fingerprintTree, this);
+  // second pass - fingerprints replaceable source code
+  this.exclude = exclude;
+  this.onlyHash = this.replaceExtensions;
+  var fingerprintTree2 = Fingerprint(assetRewrite, this);
+  var assetRewrite2 = UseRev(fingerprintTree2, this);
+
+  return assetRewrite2;
```

Comment:
I realize that this isn't new, but I _vaguely_ dislike returning another object from a constructor like this.